File: 2558-take-gifts-from-the-richest-pile/2558-take-gifts-from-the-richest-pile.java
class Solution {
    public long pickGifts(int[] gifts, int k) {
        long max=0,count=0;
        for(int i =1;i<=k;i++)
        {
            Arrays.sort(gifts);
            gifts[gifts.length-1]=(int)Math.sqrt(gifts[gifts.length-1]);
        }
        for(int i =0;i<gifts.length;i++)
            {
                count = count +gifts[i];
            }
        return count;
    }
}

File: 2558-take-gifts-from-the-richest-pile/2558-take-gifts-from-the-richest-pile.cpp
class Solution {
public:
    long long pickGifts(vector<int>& gifts, int k) {
        int n=gifts.size()-1;
        long s=0;
       for(int i=0;i<k;i++){
             sort(gifts.begin(),gifts.end());  
            gifts[n]=sqrt(gifts[n]);                   
        }
        for(int i=0;i<=n;i++)
            s+=gifts[i];
        return s;
    }
};